src/troute-network/troute/nhd_io.py

```diff
@@ -1559,3 +1559,274 @@ def lastobs_df_output(
     # write-out LastObs file as netcdf
     output_path = pathlib.Path(lastobs_output_folder + "/nudgingLastObs." + modelTimeAtOutput_str + ".nc").resolve()
     ds.to_netcdf(str(output_path))
+
+def write_waterbody_netcdf(
+    wbdy_filepath, 
+    wbdy_df, 
+    waterbodies_df, 
+    t0, 
+    dt, 
+    nts
+):
+    
+    '''
+    Write results of waterbodies to netcdf. 
+    Written as one file per time step.
+    
+    Arguments
+    -------------
+        wbdy_filepath (Path or string) - directory where files will be created
+        wbdy_df (DataFrame) - t-route waterbody inflow, outflow, and depth results
+        waterbodies_df (DataFrame) - linkIDs of waterbodies in network
+        t0 (datetime) - initial time
+        dt (int) - timestep duration (seconds)
+        nts (int) - number of timesteps in simulation
+        time_index (ind) - specific timestep for this file
+        
+    Returns
+    -------------
+    
+    '''
+    
+    # array of segment linkIDs at gage locations. Results from these segments will be written
+    waterbodies_df = waterbodies_df.reset_index().rename(columns = {'lake_id': 'ID'})[['ID','lat','lon','crs']]
+    wbdy_feature_id = waterbodies_df.ID.to_numpy(dtype = "int64")
+
+    # dataframe of waterbody types
+    wbdy_type = wbdy_df.loc[:,['ID','reservoir_type']].drop_duplicates().reservoir_type.to_numpy(dtype = 'int32')
+    
+    # array of simulated flow data at gage locations
+    wbdy_data = wbdy_df.drop(columns='reservoir_type').pivot(index=['ID','time'],columns='variable',values='value')
+    
+    # array of simulation time
+    wbdy_time = [t0 + timedelta(seconds = (wbdy_df.time.unique()[0] + 1).tolist() * dt)]
+    
+    # Merge waterbodies_df with wbdy_data
+    full_wbdy_data = pd.merge(waterbodies_df,wbdy_data,on = 'ID')
+    
+    if wbdy_filepath:
+        
+        # open netCDF4 Dataset in write mode
+        with netCDF4.Dataset(
+            filename = wbdy_filepath + str(wbdy_time[0].strftime('%Y%m%d%H%M')) + '.LAKEOUT.nc',
+            mode = 'w',
```

Comment:
not sure if it is a permissions issue, but if I re-run the code before I delete the batch of LAKEOUT files previously created, then I get this error. 

```
PermissionError: [Errno 13] Permission denied: b'lakeout/202108231300.LAKEOUT.nc'
```
We need to figure out how to over-write existing files, somehow. I know there are numerous `mode` argument options - does one of them meet our needs? If not, we might need to delete existing files before we go about re-creating them (same outcome, as far as I can tell).

Reply:
Mode 'w' is write and is supposed to overwrite the file if it already exists. I am not seeing this PermissionError when I re-run the code without deleting the existing LAKEOUT files, so perhaps it is a permission issue.

But we could try mode 'a' (append) and simply re-write all of the information to an existing file?

Reply:
Mode 'a' should also create a new file if one does not already exist.

Reply:
hm - I am no longer seeing this error either. I might have had a file open in another python instance when I re-ran the code.  Let's stick with `w`